PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRunner.java

```diff
@@ -116,9 +117,12 @@ public ISparkBatchJob buildSparkBatchJob(@NotNull SparkSubmitModel submitModel,
                     throw new ExecutionException("Error getting access token based on the given ADLS root path");
                 }
                 break;
+            case WEBHDFS:
+                webHdfsUploadPath = submitModel.getJobUploadStorageModel().getUploadPath();
+                break;
         }
 
-        return new SparkBatchJob(submitModel.getSubmissionParameter(), SparkBatchSubmission.getInstance(), ctrlSubject, storageAccount, accessToken, normalizedAdlRootPath);
+        return new SparkBatchJob(submitModel.getSubmissionParameter(), SparkBatchSubmission.getInstance(), ctrlSubject, storageAccount, accessToken, normalizedAdlRootPath, webHdfsUploadPath);
```

Comment:
@t-rufang , could you leverage SparkBatchAzureSubmission to replace SparkBatchSubmission?